### DISCUSSION TITLE
[Dev Tools] fullnode issue

### DISCUSSION THREAD

**Discussion by howardchang7414** [zkSync-Community-Hub/zksync-developers#753] [Dev Tooling] [answered]
### Environment

Mainnet

### Select the Dev Tool you are using

Local Node

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Get this error from docker fullnode around 15 min. How to fix it. 

Failed to get status changes from the main node: JSON-RPC request resolve_l1_batch_to_l2_block(number=L1BatchNumber(493517)) failed: Request timeout

**Comment by alhenaai** [comment]
It appears that your fullnode is experiencing a timeout issue when trying to resolve a batch from L1 to L2. This could be due to network latency or an overloaded node. Here are a few steps you can take to address this issue:

1. **Check Network Connection:** Ensure that your network connection is stable and robust enough to handle requests to and from the main node.
2. **Increase Timeout Settings:** Adjust the timeout settings in your node configuration to allow more time for requests to complete.
3. **Optimize Node Performance:** Utilize monitoring and management tools like Prometheus and Grafana to track the performance and health of your node. This can help identify bottlenecks or performance issues.

For more detailed guidance and to discuss this issue with other developers, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
Just to clarify which one of these is it that you're using?
1. Mainnet ZKsync Node https://docs.zksync.io/zksync-node/quickstart
2. Local Node https://docs.zksync.io/build/zksync-cli/running-a-node

Can you provide more information about how you set things up? A link to the docs you used would be helpful to reproduce the issue too.

**Reply by howardchang7414** [reply]
This one. 
1. Mainnet ZKsync Node https://docs.zksync.io/zksync-node/quickstart

**Reply by bxpana** [reply]
Has this happened every time? Or did it just happen once?

**Reply by howardchang7414** [reply]
Happened three times.

**Comment by popzxc** [comment, marked as answer]
This error is not a hard error, it's just a warning that something may be wrong with the network or configuration.
Given that the indicated problem is request timeout, it might indicate the connectivity issues. This is covered in [the troubleshooting section](https://docs.zksync.io/zksync-node/troubleshooting#logs).

If the node works well otherwise, and the metrics are healthy, there is no need to worry -- the node will automatically retry any failed requests. You can check the health of the node using [provided metrics](https://docs.zksync.io/zksync-node/observability).

I'm marking this question as answered for now. If you think it's not resolved, please feel free to open a new discussion, but please provide more context, e.g. what is the exact problem (node is not synced, it crashes, etc), what have you tried to do to resolve it, and what's your configuration.
